addons/egoventure/cache/cache_update_dialog.gd

```diff
@@ -0,0 +1,253 @@
+tool
+class_name CacheUpdateDialog
+extends WindowDialog
+
+
+# list of scenes in scene directory
+var _scene_list: Array
+var _scene_list_delta: Array
+
+# Cache map which gets generated
+var _cache_map = CacheMap.new()
+var _cache_map_time_modified
+
+
+# Show the cache update dialog popup
+func show_popup():
+	# configure buttons
+	get_close_button().hide()
+	$VBox/HBoxContainer/RunDelta.set_disabled(false)
+	$VBox/HBoxContainer/RunFull.set_disabled(false)
+	$VBox/HBoxContainer/Cancel.set_disabled(false)
+	$VBox/HBoxContainer/RunDelta.call_deferred("grab_focus")
+	
+	# read scene directory path from configuration
+	var configuration = preload("res://configuration.tres")
+	var scene_dir = configuration.cache_scene_path
+	
+	# retrieve CacheMap and last modified timestamp
+	if ResourceLoader.exists("res://cache_map.tres"):
+		_cache_map = ResourceLoader.load("res://cache_map.tres")
+		var file = File.new()
+		_cache_map_time_modified = file.get_modified_time("res://cache_map.tres")
+	else:
+		_cache_map_time_modified = 0
+	
+	# retrieve list of scenes in scene directory
+	_scene_list.clear()
+	_scene_list_delta.clear()
+	_read_scene_list(scene_dir)
+#	_scene_list = _read_scene_list(scene_dir)
+	
+	$VBox/SceneCount.text = "The project contains %s scenes. (%s of them were changed after last cache map update)" \
+		% [_scene_list.size(), _scene_list_delta.size()]
+	$VBox/ProgressBar.value = 0.0
+	self.popup_centered()
+	# resize popup to ensure that it fits to rendered nodes
+	self.set_size($VBox.get_rect().size+Vector2(20,20))
+
+
+# Start updating the cache map
```

Comment:
Add a parameter documentation for the _full parameter. Also, don't use a "\_"-prefix for that parameter